src/lib/libdylink.js

```diff
@@ -1172,6 +1172,96 @@ var LibraryDylink = {
 #endif
   },
 
+  $locateLibraryFromFS__deps: ['$FS'],
+  $locateLibraryFromFS: (filename, searchDirs, maxLength = Infinity) => {
+    // Find the library in the filesystem.
+    // returns null if not found.
+    if (typeof FS.lookupPath !== 'function') {
+      // wasmfs does not implement FS.lookupPath
+#if DYLINK_DEBUG
+      dbg("locateLibraryFromFS: FS.lookupPath not implemented");
+#endif
+      return null;
+    }
+
+    var candidates = [];
+    if (filename.charAt(0) === '/') {  // abs path
+      candidates.push(filename);
+    } else if (searchDirs) {
+      for (var dir of searchDirs) {
+        // PATH.join does not work well with symlinks
+        candidates.push(dir + '/' + filename);
+      }
+    } else {
+      return null;
+    }
+
+#if DYLINK_DEBUG
+    dbg("locateLibraryFromFS: candidates " + candidates);
+#endif
+
+    for (var path of candidates) {
+      try {
+        var res = FS.lookupPath(path);
+        if (res.node.isDir || res.node.isDevice) {
+          continue
+        }
+
+        if (res.path.length >= maxLength) {
+          continue
+        }
+#if DYLINK_DEBUG
+        dbg(`locateLibraryFromFS: found ${res.path} for (${filename})`);
+#endif
+        return res.path;
+      } catch(e) {
+#if DYLINK_DEBUG
+        dbg(`locateLibraryFromFS: ${path} not found: ${e}`);
+#endif
+        // do nothing is file is not found
+      }
+    }
+
+    return null;
+  },
+
+  $getDefaultLibDirs__deps: ['$ENV'],
+  $getDefaultLibDirs__proxy: 'sync',
+  $getDefaultLibDirs: () => {
+    var ldLibraryPath = ENV['LD_LIBRARY_PATH']
+#if DYLINK_DEBUG
+    dbg(`getDefaultLibDirs: LD_LIBRARY_PATH=${ldLibraryPath}`);
+#endif
+    return ldLibraryPath?.split(':') ?? [];
+  },
+
+  _dylink_resolve_path_js__deps: ['$UTF8ToString', '$stringToUTF8', '$locateLibraryFromFS', '$getDefaultLibDirs'],
+  _dylink_resolve_path_js__proxy: 'sync',
+  _dylink_resolve_path_js: (cbuf, cfile, buflen) => {
+    var cfilePtr = cfile;
+
+#if MEMORY64
+    cfilePtr = Number(cfilePtr);
+    buflen = Number(buflen)
+#endif
```

Comment:
So in the memory 64 case, we need the return value to be a `BigInt`? I wonder if the best thing to do would be:
```suggestion
#if MEMORY64
    cbuf = Number(cbuf);
    cfile = Number(cfile);
    buflen = Number(buflen)
#endif
```
and then explicitly convert back to `BigInt` while returning.

Reply:
But surprisingly I can't trigger any more failures with the current implementation. So I guess it's probably fine? I'll add a wasm64 variant to the LD_LIBRARY_PATH test just to be sure.